reflex/state.py

```diff
@@ -738,13 +738,43 @@ async def _process(self, event: Event) -> AsyncIterator[StateUpdate]:
             # Clean the state to prepare for the next event.
             self._clean()
 
+    def _check_valid(self, handler: EventHandler, events: Any) -> Any:
+        """Check if the events yielded are valid. They must be EventHandlers or EventSpecs.
+
+        Args:
+            handler: EventHandler.
+            events: The events to be checked.
+
+        Raises:
+            TypeError: If any of the events are not valid.
+
+        Returns:
+            The events as they are if valid.
+        """
+        if (
+            events is None
+            or (
+                isinstance(events, list)
```

Comment:
```suggestion
                isinstance(events, (list, tuple))
```

i think it's valid for users to return / yield a tuple of events as well as a list.

to avoid the explicit `isinstance` check, you might structure it something like

```python
if events is None or isinstance(events, (EventHandler, EventSpec)):
    return events

try:
    if all(isinstance(e, (EventHandler, EventSpec)) for e in events):
        return events
except TypeError:
    pass  # events is not iterable

raise TypeError(...)
```

Reply:
this is a good idea!

Reply:
I guess essentially an iterable of the valid types

Reply:
how do you feel about check for events being `Iterable`?

Reply:
is it possible to check for something being `Iterable` in a reliable way without calling `iter` on it?

Reply:
I think check for being `Iterable` looks for the definition of the iter magic function. I tried defining a class with that function defined, then `isinstance` returns True. That seems to be the same as iterate inside a try/except block?

Reply:
I realized after commenting. If it's a user defined class, the `__iter__` could be implemented wrong and still get `isinstance` to return True. 

I'll change it to the try/except way. That's definitely more reliable